### PR TITLE
add specialized size_t comparison macros to reduce number of warnings

### PR DESCRIPTION
**Pull request by Mistobaan**

### CODE REVIEW

**Comment by danmane** [comment]
@keveman Can you please take a look?

File: tensorflow/core/platform/default/logging.h [hunk 1]
```diff
@@ -173,6 +174,8 @@ string* MakeCheckOpString(const T1& v1, const T2& v2, const char* exprtext) {
 // The (int, int) specialization works around the issue that the compiler
 // will not instantiate the template version of the function on values of
 // unnamed enum type - see comment below.
+// The (size_t, int) and (int, size_t) specialization are to handle unsigned
+// comparison errors while still being through with the comparison.
```
Comment (martinwicke) [inline tensorflow/core/platform/default/logging.h]: through -> thorough

**Review by martinwicke — APPROVED** [review]
LGTM

**Review by martinwicke — CHANGES_REQUESTED** [review]
Please fix the typo though.

**Comment by Mistobaan** [comment]
@martinwicke fixed

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Review by martinwicke — APPROVED** [review]

**Pull request merged**